kv/kvserver: TestStoreRangeMergeRaftSnapshot failed

kv/kvserver.TestStoreRangeMergeRaftSnapshot [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14586669?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14586669?buildTab=artifacts#/) on master @ [87ed0b2667d8217499d89495be867facdb4e5231](https://github.com/cockroachdb/cockroach/commits/87ed0b2667d8217499d89495be867facdb4e5231):


```
=== RUN   TestStoreRangeMergeRaftSnapshot
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/e01daf5a27bf540c234d4bb071d72384/logTestStoreRangeMergeRaftSnapshot2761047857
    test_log_scope.go:81: use -show-logs to present logs inline
    client_merge_test.go:4022: 0=auxiliary/sstsnapshot/69/e9550eb9-1957-4ba9-8478-7ba7de48c98a/4.sst
    client_merge_test.go:4033: ssts not byte-wise identical: /artifacts/tmp/_tmp/e01daf5a27bf540c234d4bb071d72384/ssts1290221966/0_exp.sst and /artifacts/tmp/_tmp/e01daf5a27bf540c234d4bb071d72384/ssts1290221966/0_act.sst
    client_merge_test.go:4198: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e01daf5a27bf540c234d4bb071d72384/logTestStoreRangeMergeRaftSnapshot2761047857
--- FAIL: TestStoreRangeMergeRaftSnapshot (2.91s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeRaftSnapshot.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

kv/kvserver.TestStoreRangeMergeRaftSnapshot [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14591378?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14591378?buildTab=artifacts#/) on master @ [a06f12f0c1f4f407211428c559431d32d9a588a1](https://github.com/cockroachdb/cockroach/commits/a06f12f0c1f4f407211428c559431d32d9a588a1):


```
=== RUN   TestStoreRangeMergeRaftSnapshot
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/e01daf5a27bf540c234d4bb071d72384/logTestStoreRangeMergeRaftSnapshot3636393660
    test_log_scope.go:81: use -show-logs to present logs inline
    client_merge_test.go:4022: 0=auxiliary/sstsnapshot/69/49daf1c0-e58f-4be8-bda3-abf84d7e6ba9/4.sst
    client_merge_test.go:4033: ssts not byte-wise identical: /artifacts/tmp/_tmp/e01daf5a27bf540c234d4bb071d72384/ssts1348827007/0_exp.sst and /artifacts/tmp/_tmp/e01daf5a27bf540c234d4bb071d72384/ssts1348827007/0_act.sst
    client_merge_test.go:4198: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/e01daf5a27bf540c234d4bb071d72384/logTestStoreRangeMergeRaftSnapshot3636393660
--- FAIL: TestStoreRangeMergeRaftSnapshot (3.04s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeRaftSnapshot.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -3884,15 +3884,11 @@ func TestStoreRangeMergeRaftSnapshot(t *testing.T) {
 		}
 		keySpans := rditer.MakeReplicatedKeySpans(inSnap.Desc)
 		sstFileWriters := map[string]sstFileWriter{}
-		for i, span := range keySpans {
+		for _, span := range keySpans {
 			file := &storage.MemObject{}
 			writer := storage.MakeIngestionSSTWriter(ctx, st, file)
-			// If UseExciseForSnapshots is enabled, the snapshot won't write the last
-			// ClearRange in the sstable.
-			if i < len(keySpans)-1 || !storage.UseExciseForSnapshots.Get(&st.SV) {
-				if err := writer.ClearRawRange(span.Key, span.EndKey, true, true); err != nil {
-					return err
-				}
+			if err := writer.ClearRawRange(span.Key, span.EndKey, true /* pointKeys */, true /* rangeKeys */); err != nil {
+				return err
 			}
 			sstFileWriters[string(span.Key)] = sstFileWriter{
 				span:   span,

